### ISSUE TITLE
Много не доделок.

### ISSUE THREAD

**Issue by armleo** [gonzazoid/openhabr#1]
На главной, да и вообще везде не отображается аватар. Предлагаю добавить проверку в зависимости от пола и отображать соответсву иконку.

Не могу коментировать статьи тк Read-Only, предлагаю сделать как на хабре Read-Only + Read-Write + FULL.

Лучше поменяй иконку, а то роскомпозор за АП забанит :D.

Не работает счетчик комментариев.

Я сейчас пытаюсь найти свои файлы в которых аналогичный Хабру интерфейс, только лучше :D.

Теги не работают.

Верхней части нету.

Предлагаю изменить тип фалов типо todo,start,restart на todo.md,start.sh,restart.sh

ОООЧЕНЬ много зависимостей.

TPL лучше переместить в корень папки SRC.

СЛИШКОМ сложная системы рейтинга и кармы предлагаю сделать примерно как на хабре. Где то была статья на хабре про систему оценок.

**Comment by gonzazoid** [comment]
>Поиск будет?
будет, и не только. Я тут временно выпал - определялся с работой. Вроде стабилизировалось, начну допиливать проект со следующей недели. Недоработок не просто много - легче перечислить то, что работает.
Основной мотив - коллективная работа, но я тут как думаю - сделать минимально работающее ядро и максимально подробную документацию. Будет востребовано - народ подтянется и дорихтует. Не будет востребовано - чего тогда упираться?

 >Не могу коментировать статьи
Можешь, не работает обновление страницы/контента

>Верхней части нету.
и не будет - что там отображать? Ссылки на  гиктаймс и мегамозг? :)

>ОООЧЕНЬ много зависимостей.
Это видимо зависимости express. Кроме экспресса и pg там ничего не тянется. А без экспресса слишком многое приходится ручками делать - я изначально попробовал так - велосипед получается.

Слушай, спасибо огромное за замечания, реально буду рад любой помощи!

**Comment by armleo** [comment]
Сам работал год назад над openhabr репа еще осталась только пустая. Ищу исходники самописные но кажись они на ЖД который испортился :(

**Comment by armleo** [comment]
Можеш поменять абсолютные адреса на относительные в *.sh файлах? И сменить систему nginx на node c child process

**Comment by armleo** [comment]
Коммит круть. Пойду деплоить на серв. :)

**Comment by gonzazoid** [comment]
Да, Сергей хорошо нашинковал, спасибо ему огромное. Надо бы теперь время выбрать - пройтись по коду и добавить полноценный постинг и комменты.

**Comment by armleo** [comment]
Слегка опаздало.... но как насчет того что бы переписать проект целиком? Я сейчас слегка занят, но к началу новово месяца начну пилить свой варинт ибо хочу Mongoose использовать, а не постгре. Использовать архитектуру 
view <- viewer
               ^
               |
View->controller->service>model

**Comment by armleo** [comment]
Вообщем сделаю выложу. Чую мой жесткий говнокод...

**Comment by armleo** [comment]
Переделка отложена на начало следующего года....

**Comment by armleo** [comment]
Переделка верстки начата.

**Comment by gonzazoid** [comment]
Смотри,  я там код жестко перелопатил но еще не выложил. Погоди пока :) Сейчас освежу репу.

**Comment by armleo** [comment]
Я все еще в ожидании :)

**Comment by gonzazoid** [comment]
Я тут начал было статей писать, слегка заглохло, вот первая - https://habrahabr.ru/post/317542/ слегка пафосно, но это что бы нубы не доставали. В общем смотри - суть в том что там с одной стороны получается фреймворк - реакт на custom-element с либой сборки стейта, то есть в отличии от реакта стейт не монолитный а собирается из кусочков что позволяет его инкапсулировать в каждом elements, расшаривая наружу нужные ендпоинты.

С другой стороны - там идет генерация типов и функции доступа к api (любому,  не важно - эдакий апификатор), и скозная типизация с генерацией типов на основании swagger спеки и хранимых процедур sql [да, парсим sql по полной, ничего сложного] вкратце я тут расписал - https://github.com/lfittl/libpg_query/pull/17

В общем получается три модуля - реактоподобный фреймворк (jsx-light), апификатор (реально мощная штука, у меня уже масса примеров по работе где он облегчил бы жизнь) и сквозная типизация фронт и бэкенда.
Это я к чему - я зашиваюсь. Мне еще семью кормить и на яхту откладывать :) В общем если интересно - мой скайп gonzazoid, стукни - поболтаем, я ищу напарника с которым можно было бы это дело добить (все в опенсорс, кроме славы и галочки в портфолио ничего на этом не заработаешь)

**Comment by armleo** [comment]
Впечетляет. Давно думал, что TSX - это круто, тем более без реакта. Единственное мне всего 17. Статья понравилась. Я тоже писал свой апификатор, один JSON с описанием всего (структура датабазы, контроллеры, etc), если нужно что-то конкретное, что не реализованно 
```
"import":{
   url:"/post/create"
   file:"файл.js",
   method:"thatMethod"
}
```

А у меня так уж получилось экзамены :(.

Сам подсел на Node.js + TypeScript, пока пытался осилить front XD

**Comment by gonzazoid** [comment]
>Единственное мне всего 17

Это вообще не имеет значения :) Мне тоже когда то было 17, а тебе когда нибудь будет 39 :)

Как соберешься - скажи, если реально интересно - разрулим. Я пока на выходных со сборкой стейта разберусь, может продолжение статьи напишу.
